### PR TITLE
Document Morse-Smale Quadrangulation example

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR adds a simple Python script and a mkdocs file to document the ttk-data  MorseSmaleQuadrangulation state file. 

Enjoy,
Pierre

### CODE REVIEW

**Comment by julien-tierny** [comment]
once your PR is merged, thanks for including a link to this example in the doxygen documentation of the involved modules.
see this commit for instance:
topology-tool-kit/ttk@2b9fd4c

File: python/morseSmaleQuadrangulation.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+# create a new 'XML Unstructured Grid Reader'
+rockerArmvtu = XMLUnstructuredGridReader(FileName=["rockerArm.vtu"])
+rockerArmvtu.PointArrayStatus = ["OutputEigenFunctions"]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: I don't believe this line is necessary.

File: python/morseSmaleQuadrangulation.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+extractComponent1.Component = 83
+
+# create a new 'TTK ScalarFieldNormalizer'
+tTKScalarFieldNormalizer1 = TTKScalarFieldNormalizer(Input=extractComponent1)
+tTKScalarFieldNormalizer1.ScalarField = ["POINTS", "Result"]
+
+# create a new 'TTK PersistenceDiagram'
+tTKPersistenceDiagram1 = TTKPersistenceDiagram(Input=tTKScalarFieldNormalizer1)
+tTKPersistenceDiagram1.ScalarField = ["POINTS", "Result"]
+tTKPersistenceDiagram1.InputOffsetField = [None, ""]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: I don't believe this line is necessary.

File: python/morseSmaleQuadrangulation.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+threshold1 = Threshold(Input=tTKPersistenceDiagram1)
+threshold1.Scalars = ["CELLS", "Persistence"]
+threshold1.ThresholdRange = [0.001, 0.9999999403953552]
+
+# create a new 'TTK TopologicalSimplification'
+tTKTopologicalSimplification1 = TTKTopologicalSimplification(
+    Domain=tTKScalarFieldNormalizer1, Constraints=threshold1
+)
+tTKTopologicalSimplification1.ScalarField = ["POINTS", "Result"]
+tTKTopologicalSimplification1.InputOffsetField = [None, ""]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: I don't believe this line is necessary.

File: python/morseSmaleQuadrangulation.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+tTKTopologicalSimplification1.VertexIdentifierField = [None, ""]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: I don't believe this line is necessary.

File: python/morseSmaleQuadrangulation.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex1 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification1)
+tTKMorseSmaleComplex1.ScalarField = ["POINTS", "Result"]
+tTKMorseSmaleComplex1.OffsetField = ["POINTS", "Result"]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: I don't believe this line is necessary.

File: python/morseSmaleQuadrangulation.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+# create a new 'TTK GeometrySmoother'
+tTKGeometrySmoother1 = TTKGeometrySmoother(Input=OutputPort(tTKMorseSmaleComplex1, 1))
+tTKGeometrySmoother1.IterationNumber = 5
+tTKGeometrySmoother1.InputMaskField = [None, ""]
+
+# create a new 'Extract Surface'
+extractSurface1 = ExtractSurface(Input=tTKGeometrySmoother1)
+
+# create a new 'Tube'
+tube1 = Tube(Input=extractSurface1)
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+tube1.Scalars = ["POINTS", "CellDimension"]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+tube1.Vectors = [None, ""]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+tube1.Radius = 0.004
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKMorseSmaleComplex1)
+tTKIcospheresFromPoints1.Radius = 0.01
+
+# create a new 'TTK GeometrySmoother'
+tTKGeometrySmoother1 = TTKGeometrySmoother(Input=OutputPort(tTKMorseSmaleComplex1, 1))
+tTKGeometrySmoother1.IterationNumber = 5
+tTKGeometrySmoother1.InputMaskField = [None, ""]
+
+# create a new 'Extract Surface'
+extractSurface1 = ExtractSurface(Input=tTKGeometrySmoother1)
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+# create a new 'Extract Surface'
+extractSurface2 = ExtractSurface(Input=tTKIdentifierRandomizer1)
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKMorseSmaleComplex1)
+tTKIcospheresFromPoints1.Radius = 0.01
+
+# create a new 'TTK GeometrySmoother'
+tTKGeometrySmoother1 = TTKGeometrySmoother(Input=OutputPort(tTKMorseSmaleComplex1, 1))
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+tTKGeometrySmoother1.IterationNumber = 5
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+tTKGeometrySmoother1.InputMaskField = [None, ""]
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+    Input=OutputPort(tTKMorseSmaleComplex1, 3),
+)
+tTKIdentifierRandomizer1.ScalarField = ["POINTS", "MorseSmaleManifold"]
+
+# create a new 'Extract Surface'
+extractSurface2 = ExtractSurface(Input=tTKIdentifierRandomizer1)
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKMorseSmaleComplex1)
+tTKIcospheresFromPoints1.Radius = 0.01
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    registrationName="TTKIdentifierRandomizer1",
+    Input=OutputPort(tTKMorseSmaleComplex1, 3),
+)
+tTKIdentifierRandomizer1.ScalarField = ["POINTS", "MorseSmaleManifold"]
+
+# create a new 'Extract Surface'
+extractSurface2 = ExtractSurface(Input=tTKIdentifierRandomizer1)
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints1 = TTKIcospheresFromPoints(Input=tTKMorseSmaleComplex1)
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: pipeline dead leaf, please remove.

File: python/morseSmaleQuadrangulation.py [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+tube1.Scalars = ["POINTS", "CellDimension"]
+tube1.Vectors = [None, ""]
+tube1.Radius = 0.004
+
+# create a new 'Generate Surface Normals'
+generateSurfaceNormals1 = GenerateSurfaceNormals(Input=extractSurface2)
+
+# create a new 'TTK MorseSmaleQuadrangulation'
+tTKMorseSmaleQuadrangulation1 = TTKMorseSmaleQuadrangulation(
+    Triangulatedsurface=tTKIdentifierRandomizer1,
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: in principle, `OutputPort(tTKMorseSmaleComplex1, 3)` could be used instead here.
the randomizer is not really needed in this python script.

File: python/morseSmaleQuadrangulation.py [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+tTKTopologicalSimplification1.InputOffsetField = [None, ""]
+tTKTopologicalSimplification1.VertexIdentifierField = [None, ""]
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex1 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification1)
+tTKMorseSmaleComplex1.ScalarField = ["POINTS", "Result"]
+tTKMorseSmaleComplex1.OffsetField = ["POINTS", "Result"]
+
+# create a new 'TTK IdentifierRandomizer'
+tTKIdentifierRandomizer1 = TTKIdentifierRandomizer(
```
Comment (julien-tierny) [inline python/morseSmaleQuadrangulation.py]: to remove (see my comment below)

File: docs/morseSmaleQuadrangulation.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+<!--[![Morse-Smale Quadrangulation example video tutorial](https://topology-tool-kit.github.io/img/gallery/morseSmaleQuadrangulation.jpg)](https://www.youtube.com/watch?v=eNW8l5PlgpU)-->
+
+<iframe width="100%" height="420"
+src="https://www.youtube.com/embed/eNW8l5PlgpU" frameborder="0"
+allowfullscreen></iframe>
+
+## Pipeline description
+
+This example first loads a mecanical model as a 2D triangle mesh from
+disk. This mechanical model embeds a vector field with the output of
```
Comment (julien-tierny) [inline docs/morseSmaleQuadrangulation.md]: a vector field -› a collection of scalar fields
(usually, a vector field on a 2D surface is understood to have only 3 components, x y and z)

**Review by julien-tierny — CHANGES_REQUESTED** [review]
thanks a lot pierre.
if you could address the few minor remarks I entered, that'd be wonderful.
thanks!

**Comment by pierre-guillou** [comment]
Everything should be good now

**Review by julien-tierny — APPROVED** [review]

**Comment by julien-tierny** [comment]
awesome, thanks!

**Pull request merged**